protocol/binary/reader.go

```diff
@@ -202,6 +207,19 @@ func (br *Reader) read(bs []byte, off int64) (int64, error) {
 	return off, err
 }
 
+// copyN copies the given number of bytes starting at the given position into
+// the Writer.
+func (br *Reader) copyN(w io.Writer, off int64, n int64) (int64, error) {
```

Comment:
I think the comment on this function confused me. Suggest taking `io.CopyN`'s lead with something like:

```
// CopyN copies n bytes (or until an error) from ...
```